File: clients/node/sdk/models/poll.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Poll = void 0;
const createEmojiFromDiscriminatorValue_1 = require("./createEmojiFromDiscriminatorValue");
const createPollOptionFromDiscriminatorValue_1 = require("./createPollOptionFromDiscriminatorValue");
class Poll {
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    get additionalData() {
        return this._additionalData;
    }
    ;
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    set additionalData(value) {
        this._additionalData = value;
    }
    ;
    /**
     * Instantiates a new Poll and sets the default values.
     */
    constructor() {
        this._additionalData = {};
    }
    ;
    /**
     * Gets the emojis property value. The emojis property
     * @returns a Emoji
     */
    get emojis() {
        return this._emojis;
    }
    ;
    /**
     * Sets the emojis property value. The emojis property
     * @param value Value to set for the emojis property.
     */
    set emojis(value) {
        this._emojis = value;
    }
    ;
    /**
     * Gets the expired property value. The expired property
     * @returns a boolean
     */
    get expired() {
        return this._expired;
    }
    ;
    /**
     * Sets the expired property value. The expired property
     * @param value Value to set for the expired property.
     */
    set expired(value) {
        this._expired = value;
    }
    ;
    /**
     * Gets the expires_at property value. The expires_at property
     * @returns a string
     */
    get expires_at() {
        return this._expires_at;
    }
    ;
    /**
     * Sets the expires_at property value. The expires_at property
     * @param value Value to set for the expires_at property.
     */
    set expires_at(value) {
        this._expires_at = value;
    }
    ;
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    getFieldDeserializers() {
        return {
            "emojis": n => { this.emojis = n.getCollectionOfObjectValues(createEmojiFromDiscriminatorValue_1.createEmojiFromDiscriminatorValue); },
            "expired": n => { this.expired = n.getBooleanValue(); },
            "expires_at": n => { this.expires_at = n.getStringValue(); },
            "id": n => { this.id = n.getStringValue(); },
            "multiple": n => { this.multiple = n.getBooleanValue(); },
            "options": n => { this.options = n.getCollectionOfObjectValues(createPollOptionFromDiscriminatorValue_1.createPollOptionFromDiscriminatorValue); },
            "own_votes": n => { this.own_votes = n.getCollectionOfPrimitiveValues(); },
            "voted": n => { this.voted = n.getBooleanValue(); },
            "voters_count": n => { this.voters_count = n.getNumberValue(); },
            "votes_count": n => { this.votes_count = n.getNumberValue(); },
        };
    }
    ;
    /**
     * Gets the id property value. The id property
     * @returns a string
     */
    get id() {
        return this._id;
    }
    ;
    /**
     * Sets the id property value. The id property
     * @param value Value to set for the id property.
     */
    set id(value) {
        this._id = value;
    }
    ;
    /**
     * Gets the multiple property value. The multiple property
     * @returns a boolean
     */
    get multiple() {
        return this._multiple;
    }
    ;
    /**
     * Sets the multiple property value. The multiple property
     * @param value Value to set for the multiple property.
     */
    set multiple(value) {
        this._multiple = value;
    }
    ;
    /**
     * Gets the options property value. The options property
     * @returns a PollOption
     */
    get options() {
        return this._options;
    }
    ;
    /**
     * Sets the options property value. The options property
     * @param value Value to set for the options property.
     */
    set options(value) {
        this._options = value;
    }
    ;
    /**
     * Gets the own_votes property value. The own_votes property
     * @returns a integer
     */
    get own_votes() {
        return this._own_votes;
    }
    ;
    /**
     * Sets the own_votes property value. The own_votes property
     * @param value Value to set for the own_votes property.
     */
    set own_votes(value) {
        this._own_votes = value;
    }
    ;
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    serialize(writer) {
        if (!writer)
            throw new Error("writer cannot be undefined");
        writer.writeCollectionOfObjectValues("emojis", this.emojis);
        writer.writeBooleanValue("expired", this.expired);
        writer.writeStringValue("expires_at", this.expires_at);
        writer.writeStringValue("id", this.id);
        writer.writeBooleanValue("multiple", this.multiple);
        writer.writeCollectionOfObjectValues("options", this.options);
        writer.writeCollectionOfPrimitiveValues("own_votes", this.own_votes);
        writer.writeBooleanValue("voted", this.voted);
        writer.writeNumberValue("voters_count", this.voters_count);
        writer.writeNumberValue("votes_count", this.votes_count);
        writer.writeAdditionalData(this.additionalData);
    }
    ;
    /**
     * Gets the voted property value. The voted property
     * @returns a boolean
     */
    get voted() {
        return this._voted;
    }
    ;
    /**
     * Sets the voted property value. The voted property
     * @param value Value to set for the voted property.
     */
    set voted(value) {
        this._voted = value;
    }
    ;
    /**
     * Gets the voters_count property value. The voters_count property
     * @returns a integer
     */
    get voters_count() {
        return this._voters_count;
    }
    ;
    /**
     * Sets the voters_count property value. The voters_count property
     * @param value Value to set for the voters_count property.
     */
    set voters_count(value) {
        this._voters_count = value;
    }
    ;
    /**
     * Gets the votes_count property value. The votes_count property
     * @returns a integer
     */
    get votes_count() {
        return this._votes_count;
    }
    ;
    /**
     * Sets the votes_count property value. The votes_count property
     * @param value Value to set for the votes_count property.
     */
    set votes_count(value) {
        this._votes_count = value;
    }
    ;
}
exports.Poll = Poll;

File: clients/csharp/sdk/Models/Poll.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System;
namespace MastodonClientLib.Models {
    /// <summary>
    /// {id}: Id of the poll{expires_at}: Date and time when the poll expires{expired}: Whether the poll has expired{multiple}: Whether the poll allows multiple choices{votes_count}: Number of votes for the poll{voters_count}: Number of voters for the poll{options}: Options for the poll{emojis}: Emojis for the poll{voted}: Whether the authorized user has voted for the poll{own_votes}: Indexes of options voted for by the authorized user
    /// </summary>
    public class Poll : IParsable {
        /// <summary>The emojis property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<Emoji>? Emojis { get; set; }
#nullable restore
#else
        public List<Emoji> Emojis { get; set; }
#endif
        /// <summary>The expired property</summary>
        public bool? Expired { get; set; }
        /// <summary>The expires_at property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? ExpiresAt { get; set; }
#nullable restore
#else
        public string ExpiresAt { get; set; }
#endif
        /// <summary>The id property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Id { get; set; }
#nullable restore
#else
        public string Id { get; set; }
#endif
        /// <summary>The multiple property</summary>
        public bool? Multiple { get; set; }
        /// <summary>The options property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<PollOption>? Options { get; set; }
#nullable restore
#else
        public List<PollOption> Options { get; set; }
#endif
        /// <summary>The own_votes property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<int?>? OwnVotes { get; set; }
#nullable restore
#else
        public List<int?> OwnVotes { get; set; }
#endif
        /// <summary>The voted property</summary>
        public bool? Voted { get; set; }
        /// <summary>The voters_count property</summary>
        public int? VotersCount { get; set; }
        /// <summary>The votes_count property</summary>
        public int? VotesCount { get; set; }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static Poll CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new Poll();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"emojis", n => { Emojis = n.GetCollectionOfObjectValues<Emoji>(Emoji.CreateFromDiscriminatorValue)?.ToList(); } },
                {"expired", n => { Expired = n.GetBoolValue(); } },
                {"expires_at", n => { ExpiresAt = n.GetStringValue(); } },
                {"id", n => { Id = n.GetStringValue(); } },
                {"multiple", n => { Multiple = n.GetBoolValue(); } },
                {"options", n => { Options = n.GetCollectionOfObjectValues<PollOption>(PollOption.CreateFromDiscriminatorValue)?.ToList(); } },
                {"own_votes", n => { OwnVotes = n.GetCollectionOfPrimitiveValues<int?>()?.ToList(); } },
                {"voted", n => { Voted = n.GetBoolValue(); } },
                {"voters_count", n => { VotersCount = n.GetIntValue(); } },
                {"votes_count", n => { VotesCount = n.GetIntValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteCollectionOfObjectValues<Emoji>("emojis", Emojis);
            writer.WriteBoolValue("expired", Expired);
            writer.WriteStringValue("expires_at", ExpiresAt);
            writer.WriteStringValue("id", Id);
            writer.WriteBoolValue("multiple", Multiple);
            writer.WriteCollectionOfObjectValues<PollOption>("options", Options);
            writer.WriteCollectionOfPrimitiveValues<int?>("own_votes", OwnVotes);
            writer.WriteBoolValue("voted", Voted);
            writer.WriteIntValue("voters_count", VotersCount);
            writer.WriteIntValue("votes_count", VotesCount);
        }
    }
}
